File: tests/fixtures/parsers/user_service.go
package service

import (
    "context"
    "errors"
)

type UserService struct {
    repo UserRepository
}

func NewUserService(repo UserRepository) *UserService {
    return &UserService{repo: repo}
}

func (s *UserService) GetUser(ctx context.Context, id int64) (*User, error) {
    if id <= 0 {
        return nil, errors.New("invalid id")
    }
    return s.repo.FindByID(ctx, id)
}

File: tests/fixtures/parsers/user_service.rs
use std::sync::Arc;

pub struct UserService {
    repo: Arc<dyn UserRepository>,
}

impl UserService {
    pub fn new(repo: Arc<dyn UserRepository>) -> Self {
        Self { repo }
    }

    pub async fn get_user(&self, id: i64) -> Result<User, Error> {
        self.repo.find_by_id(id).await
    }

    fn internal_helper(&self) -> bool {
        true
    }
}
